app/controllers/admin/bulk_republishing_controller.rb

```diff
@@ -79,6 +79,81 @@ def republish_documents_by_organisation
     end
   end
 
+  def new_documents_by_content_ids; end
+
+  def search_documents_by_content_ids
+    content_ids = content_ids_string_to_array(params[:content_ids])
+
+    if content_ids.empty?
+      flash[:alert] = "No content IDs provided"
+      return redirect_to(admin_bulk_republishing_documents_by_content_ids_new_path)
+    end
+
+    matching_document_content_ids = Document.where(content_id: content_ids).pluck(:content_id)
+
+    unless matching_document_content_ids.count == content_ids.count
+      unfound_content_ids = content_ids - matching_document_content_ids
+      flash[:alert] = "Unable to find document(s) with the following content IDs: #{content_ids_array_to_string(unfound_content_ids)}"
+      return redirect_to(admin_bulk_republishing_documents_by_content_ids_new_path)
+    end
+
+    redirect_to(admin_bulk_republishing_documents_by_content_ids_confirm_path(params[:content_ids]))
+  end
+
+  def confirm_documents_by_content_ids
+    @content_ids_string = params[:content_ids]
+    content_ids = content_ids_string_to_array(@content_ids_string)
+
+    if content_ids.empty?
+      flash[:alert] = "No content IDs provided"
+      return redirect_to(admin_bulk_republishing_documents_by_content_ids_new_path)
+    end
+
+    @matching_documents = Document.where(content_id: content_ids)
+
+    unless @matching_documents.count == content_ids.count
+      unfound_content_ids = content_ids - @matching_documents.pluck(:content_id)
+      flash[:alert] = "Unable to find document(s) with the following content IDs: #{content_ids_array_to_string(unfound_content_ids)}"
+      return redirect_to(admin_bulk_republishing_documents_by_content_ids_new_path)
+    end
+
+    @republishing_event = RepublishingEvent.new
+  end
+
+  def republish_documents_by_content_ids
+    @content_ids_string = params[:content_ids]
+    content_ids = content_ids_string_to_array(@content_ids_string)
+
+    if content_ids.empty?
+      flash[:alert] = "No content IDs provided"
+      return redirect_to(admin_bulk_republishing_documents_by_content_ids_new_path)
+    end
+
+    # we could pluck just the IDs and content IDs here for speed, but this is more readable
+    @matching_documents = Document.where(content_id: content_ids)
+
+    unless @matching_documents.count == content_ids.count
```

Comment:
Could this be extracted to a private method as we're using it a few times here?

Reply:
Something like [this](https://github.com/alphagov/whitehall/pull/9170/commits/8ae0b5db4b735d46eb372dcaca9fbd55a5e4d7c7)?

Reply:
Yeah - is there a way we could remove the `return if` and just call `redirect_if` or do we need to do it this way to allow us to exit out of the controller action early?

Reply:
Pretty sure we'd have a multiple render error without it, since the code would reach multiple `redirect_to` or `render` statements